### PR TITLE
docs: update doc to reflect recent api changes

### PR DESCRIPTION
**Pull request by jingwei99**
Updates the [Start Envoy](https://github.com/lyft/envoy-mobile/blob/main/docs/root/api/starting_envoy.rst) section and adds doc for `Stats` to reflect the api changes introduced by https://github.com/lyft/envoy-mobile/commit/b9a4334916159eebe68a544690a1eb9e4023dbc4

Signed-off-by: Jingwei Hao <jingweih@lyft.com>

### CODE REVIEW

File: docs/root/api/starting_envoy.rst [hunk 1]
```diff
@@ -7,32 +7,34 @@ Starting Envoy
...
+  let streamClient = try EngineBuilder()
     .addLogLevel(.warn)
     ...
     .build()
+    .streamClient()
 
 -----------------------
-``StreamClientBuilder``
+``EngineBuilder``
 -----------------------
```
Comment (buildbreaker) [inline docs/root/api/starting_envoy.rst:34]: Formatting issues

File: docs/root/api/stats.rst [hunk 2]
```diff
@@ -0,0 +1,22 @@
+.. _api_stats:
+
+Stats
+=========================
```
Comment (buildbreaker) [inline docs/root/api/stats.rst:4]: Same here

File: docs/root/api/stats.rst [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+----------------
+``StatsClient``
+----------------
```
Comment (buildbreaker) [inline docs/root/api/stats.rst:8]:
```suggestion
---------------
``StatsClient``
---------------
```
Looks like it's one character off

**Comment by jingwei99** [comment]
> Have you tried publishing this locally to make sure it looks like what you expect?
> 
> The docs for doing that are here: https://github.com/lyft/envoy-mobile/tree/ef442f42ff257c94ecc3f6cc3c9115567e45d2fe/docs#generating-docs-locally

yup looks good locally actually, i'll remove the extra symbols

File: docs/root/api/stats.rst [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+---------------
+``StatsClient``
+---------------
+
+To use Envoy Mobile's stats function, obtain an instance of ``StatsClient`` from ``Engine`` (refer to :ref:`api_starting_envoy` for building an engine instance), and store the stats client to create ``Counter`` instances.
+
+**Kotlin example**::
+
+  statsClient.counter(Element("foo"), Element"bar"))
```
Comment (buildbreaker) [inline docs/root/api/stats.rst:14]:
```suggestion
  statsClient.counter(Element("foo"), Element("bar"))
```

File: docs/root/api/stats.rst [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+**Swift example**::
+
+  statsClient.counter(elements: ["foo", "bar"])
+
+The ``counter`` method from stats client takes a variable number of elements and returns a ``Counter`` instance, the elements are used to for a dot(.) delimited string. For the example code above, the formed string is ``foo.bar``, this string serves as the identifier of the counter.
+
+Store the counter instance, and call the ``increment`` method to increment the counter wherever it applies.
```
Comment (buildbreaker) [inline docs/root/api/stats.rst:22]: Could we have examples on this as well?

**Pull request merged**